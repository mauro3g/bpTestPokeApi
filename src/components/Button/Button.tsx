import React from 'react'

const style: { [key: string]: React.CSSProperties } = {
    button: {
        color: 'white',
        textAlign: 'center',
        fontSize: '1rem',
        padding: '10px 32px',
        margin: '2px 2px',
        cursor: 'pointer',
        border: '2px solid #008CBA',
        backgroundColor: '#008CBA',
        borderRadius: '5%'
    },
} as const

interface Props {
    type: "button" | "submit" | "reset" | undefined,
    label: string
}

const Button = (props: Props) => {
    const { type, label } = props
    return (
        <button style={style.button} type={type}>{label}</button>
    )
}

export default Button