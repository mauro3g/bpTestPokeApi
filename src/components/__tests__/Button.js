import { render, screen } from '@testing-library/react';
import Button from '../Button/Button'
import renderer from 'react-test-renderer'

test('should render a button component', () => {
    render(<Button type="submit" label={"testbtn"}></Button>)
    const element = screen.getByTestId('button-component')
    expect(element).toBeInTheDocument()
    expect(element).toHaveTextContent("testbtn")
})

test('match snapshop', ()=>{
    const testBtn = renderer.create(<Button type="submit" label={"testbtn"}></Button>).toJSON()
    expect(testBtn).toMatchSnapshot()
})