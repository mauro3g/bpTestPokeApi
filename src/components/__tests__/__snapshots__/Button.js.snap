// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshop 1`] = `
<button
  data-testid="button-component"
  style={
    Object {
      "backgroundColor": "#008CBA",
      "border": "2px solid #008CBA",
      "borderRadius": "5%",
      "color": "white",
      "cursor": "pointer",
      "fontSize": "1rem",
      "margin": "2px 2px",
      "padding": "10px 32px",
      "textAlign": "center",
    }
  }
  type="submit"
>
  testbtn
</button>
`;
